From 7ba4151a499e5646f1ae405345da12ea16b5b66f Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Thu, 29 Apr 2021 11:29:46 -0700
Subject: [PATCH 088/196] Update network client name to "lotusd" from "Bitcoin
 ABC" (#162)

Make the client advertise itself as "lotusd" in the user agent in the version message to other peers.

"lotusd" leaves room for other implementations to join in with different user agent string.
---
 src/clientversion.cpp                     | 2 +-
 src/test/net_tests.cpp                    | 2 +-
 test/functional/abc-cmdline.py            | 2 +-
 test/functional/abc_rpc_excessiveblock.py | 6 +++---
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/clientversion.cpp b/src/clientversion.cpp
index 536b8ac7c..287dfed5b 100644
--- a/src/clientversion.cpp
+++ b/src/clientversion.cpp
@@ -11,7 +11,7 @@
  * for both bitcoind and bitcoin-qt, to make it harder for attackers to
  * target servers or GUI users specifically.
  */
-const std::string CLIENT_NAME("Bitcoin ABC");
+const std::string CLIENT_NAME("lotusd");
 
 #ifdef HAVE_BUILD_INFO
 #include <obj/build.h>
diff --git a/src/test/net_tests.cpp b/src/test/net_tests.cpp
index 53cce1625..8240837e4 100644
--- a/src/test/net_tests.cpp
+++ b/src/test/net_tests.cpp
@@ -220,7 +220,7 @@ BOOST_AUTO_TEST_CASE(test_userAgent) {
     gArgs.ForceSetMultiArg("-uacomment", {uacomment});
 
     const std::string versionMessage =
-        "/Bitcoin ABC:" + ToString(CLIENT_VERSION_MAJOR) + "." +
+        "/lotusd:" + ToString(CLIENT_VERSION_MAJOR) + "." +
         ToString(CLIENT_VERSION_MINOR) + "." +
         ToString(CLIENT_VERSION_REVISION) + "(EB8.0; " + uacomment + ")/";
 
diff --git a/test/functional/abc-cmdline.py b/test/functional/abc-cmdline.py
index d7d279c9a..9c445f1e5 100755
--- a/test/functional/abc-cmdline.py
+++ b/test/functional/abc-cmdline.py
@@ -48,7 +48,7 @@ class ABC_CmdLine_Test (BitcoinTestFramework):
             2 * LEGACY_MAX_BLOCK_SIZE)])
         self.check_excessive(2 * LEGACY_MAX_BLOCK_SIZE)
         # Check for EB correctness in the subver string
-        self.check_subversion(r"/Bitcoin ABC:.*\(EB2\.0; .*\)/")
+        self.check_subversion(r"/lotusd:.*\(EB2\.0; .*\)/")
 
         self.log.info("  Attempt to set below legacy limit of 1MB - try {} bytes".format(
             LEGACY_MAX_BLOCK_SIZE))
diff --git a/test/functional/abc_rpc_excessiveblock.py b/test/functional/abc_rpc_excessiveblock.py
index 30b14b53b..bcfd7a86e 100755
--- a/test/functional/abc_rpc_excessiveblock.py
+++ b/test/functional/abc_rpc_excessiveblock.py
@@ -66,7 +66,7 @@ class ExcessiveBlockSizeRPCTest(BitcoinTestFramework):
         ebs = getsize['excessiveBlockSize']
         assert_equal(ebs, 2 * ONE_MEGABYTE)
         # Check for EB correctness in the subver string
-        self.check_subversion(r"/Bitcoin ABC:.*\(EB2\.0; .*\)/")
+        self.check_subversion(r"/lotusd:.*\(EB2\.0; .*\)/")
 
         # Check setting to 13MB
         self.nodes[0].setexcessiveblock(13 * ONE_MEGABYTE)
@@ -74,7 +74,7 @@ class ExcessiveBlockSizeRPCTest(BitcoinTestFramework):
         ebs = getsize['excessiveBlockSize']
         assert_equal(ebs, 13 * ONE_MEGABYTE)
         # Check for EB correctness in the subver string
-        self.check_subversion(r"/Bitcoin ABC:.*\(EB13\.0; .*\)/")
+        self.check_subversion(r"/lotusd:.*\(EB13\.0; .*\)/")
 
         # Check setting to 13.14MB
         self.nodes[0].setexcessiveblock(13140000)
@@ -82,7 +82,7 @@ class ExcessiveBlockSizeRPCTest(BitcoinTestFramework):
         ebs = getsize['excessiveBlockSize']
         assert_equal(ebs, 13.14 * ONE_MEGABYTE)
         # check for EB correctness in the subver string
-        self.check_subversion(r"/Bitcoin ABC:.*\(EB13\.1; .*\)/")
+        self.check_subversion(r"/lotusd:.*\(EB13\.1; .*\)/")
 
     def run_test(self):
         self.test_excessiveblock()
-- 
2.39.5 (Apple Git-154)

